api_spec.yaml

```diff
@@ -2532,6 +2587,29 @@ components:
                 - product
                 - name
                 - description
+        coreConfigType:
+            type: object
+            properties:
+                name:
+                    type: string
+                    description: Name of the property
+                defaultValue:
+                    oneOf:
+                        - type: string
+                        - type: boolean
+                        - type: number
+                    nullable: true
+                    description: Default value of the property
+                description:
+                    type: string
+                    description: Description of the property
+                differentAcrossTenants:
+                    type: boolean
+                    description: True if the property is different across tenants
```

Comment:
```suggestion
```

We should not be needing this because we should only include properties that are different across tenants in this list

Reply:
What about public tenant though we would be showing all the properties for that tenant right?

Reply:
Right, I guess we keep it then. Please make the description more clear. True means that you can set different values per tenant.

Reply:
There is also a concept of protected property. In the SaaS environment, the user is not allowed to change db settings (protected props). Also it won't be returned by the core.

https://github.com/supertokens/supertokens-postgresql-plugin/blob/033c015d17793b79c0f033c13641b5fd238b122a/src/main/java/io/supertokens/storage/postgresql/Start.java#L116

https://github.com/supertokens/supertokens-core/blob/c1800abcce12247f67fc0699498561613826aa7f/src/main/java/io/supertokens/config/CoreConfig.java#L47

Reply:
> Please make the description more clear.

Updated the description.

> the user is not allowed to change db settings (protected props). Also it won't be returned by the core.

Okay what are we supposed to do in that case since it isn't returned by the core

Reply:
I guess we don't need to do anything now. The core will throw an error anyways if the user tries to update them